### PR TITLE
News edit functionality

### PR DESCRIPTION
**Pull request by aumu1031**

### CODE REVIEW

File: classes/news/NewsAPIPresenter.php [hunk 1]
```diff
@@ -22,7 +22,32 @@ function (NewsPostDto $post) use ($system) {
...
+            $newsManager->getNewsPosts($num_posts)
+        );
+    }
+
+    /**
+     * @param NewsPostDto[] $newsPosts
+     * @return array
+     */
+    public static function savePostResponse(NewsManager $newsManager, System $system, int $post_id, string $title, string $version, string $content, bool $update, bool $bugfix, bool $event, int $num_posts = 8): array {
+            $newsManager->saveNewsPost($post_id, $title, $version, $content, $update, $bugfix, $event);
```
Comment (levimeahan) [inline classes/news/NewsAPIPresenter.php:34]: We should not have the presenter be changing any data, can you move this `saveNewsPost` call up to `api/news.php`?

File: ui_components/src/home/Home.css [hunk 2]
```diff
@@ -800,9 +800,23 @@ body {
     margin-left: 5px;
     margin-right: 5px;
 }
-.news_item_details {
+.news_item_edit {
+    font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
+    font-size: 12px;
+    font-weight: bold;
+    color: #6a5e46;
```
Comment (levimeahan) [inline ui_components/src/home/Home.css:807]: TODO at some point - extract these colors to vars so we could more easily re-theme the logged out view if desired

File: ui_components/src/home/Home.js [hunk 3]
```diff
@@ -635,5 +665,75 @@ function FooterSection({ }) {
...
+                </radialGradient>
+                <rect className="register_button_background" width="100%" height="100%" />
+                <text className="register_button_shadow_text" x="81" y="18" textAnchor="middle" dominantBaseline="middle">create a character</text>
+                <text className="register_button_text" x="81" y="16" textAnchor="middle" dominantBaseline="middle">create a character</text>
+            </svg>
+        </>
+    );
+}
+
+function LogoutButtons({ homeLinks }) {
```
Comment (levimeahan) [inline ui_components/src/home/Home.js:701]: Would this be more accurate? 
```suggestion
function LoggedOutButtons({ homeLinks }) {
```

File: ui_components/src/home/Home.js [hunk 4]
```diff
@@ -559,6 +511,36 @@ function NewsSection({ newsRef, newsPosts, homeLinks }) {
         return formattedDate;
     }
 
+    function cleanNewsContents(contents) {
+        console.log(contents);
+        const parser = new DOMParser();
+        const decodedString = parser.parseFromString(contents.replace(/[\r\n]+/g, " ").replace(/<br\s*\/?>/g, '\n'), 'text/html').body.textContent;
+        return decodedString;
```
Comment (levimeahan) [inline ui_components/src/home/Home.js:518]: Can you add a comment about what the intent of this is?

**Review by levimeahan — APPROVED** [review]

File: ui_components/src/home/Home.js [hunk 5]
```diff
@@ -559,6 +511,36 @@ function NewsSection({ newsRef, newsPosts, homeLinks }) {
...
+    }
+
+    function saveNewsItem(postId) {
+        console.log(contentRef.current.value);
```
Comment (levimeahan) [inline ui_components/src/home/Home.js:522]: Could you pull these `console.log` calls into a `debug` function so we can have it off by default, but turn on with a console var or in dev?

Example:
https://github.com/elementum-games/shinobi-chronicles/blob/46849f675f6235bd65c2cdf0bbaf5b5c48f5a3ca/ui_components/src/travel/Travel.js#L60-L64

File: api/news.php [hunk 6]
```diff
@@ -30,6 +30,25 @@
...
+                'postData' => NewsAPIPresenter::newsPostResponse($NewsManager, $system, $num_posts),
+            ];
+            break;
+        case "saveNewsPost":
+            $post_id = $system->db->clean($_POST['post_id']);
+            $title = $system->db->clean($_POST['title']);
+            $version = $system->db->clean($_POST['version']);
+            $content = $system->db->clean($_POST['content']);
+            $update = $system->db->clean($_POST['update']) === "true";
+            $bugfix = $system->db->clean($_POST['bugfix']) === "true";
```
[the remark this answers is not in the document]
Comment (aumu1031) [reply api/news.php]: Fixed in commit.

File: classes/news/NewsAPIPresenter.php (hunk 1 above)
Comment (aumu1031) [reply classes/news/NewsAPIPresenter.php]: Fixed in commit.

File: ui_components/src/home/Home.js (hunk 3 above)
Comment (aumu1031) [reply ui_components/src/home/Home.js:701]: Will fix ambiguous name in follow-up.

File: ui_components/src/home/Home.js (hunk 4 above)
Comment (aumu1031) [reply ui_components/src/home/Home.js:518]: Per discord convo.

File: ui_components/src/home/Home.css (hunk 2 above)
Comment (aumu1031) [reply ui_components/src/home/Home.css:807]: Marking as to-do.

File: ui_components/src/home/Home.js (hunk 5 above)
Comment (aumu1031) [reply ui_components/src/home/Home.js:522]: Addressing in follow-up PR.

**Pull request merged**